### DISCUSSION TITLE
Improve bundle size, import strategies, separate `menu` import

### DISCUSSION THREAD

**Discussion by luwes** [muxinc/media-chrome#875] [Proposals]
The substantial increase in bundle size (~25%) after merging all the menu related components made us open a discussion around this.

- [ ] Internally we've agreed to extract the menu related components in a separate import to keep the overall default media-chrome bundle leaner. This will require an extra import in the future, for example:
```js
import 'media-chrome';
import 'media-chrome/dist/menu.js';
```
- [ ] Documentation of import strategies should be improved. 
Things around cherry-picking modules, maybe a note on jsdelivr and /+esm, possibly import maps etc.
- [ ] Spike on improving bundle size with minifying HTML/CSS, DRY code, code golfing, etc.

The largest gains in improving bundle size are still on the video engine side so this is not a very high priority.
Feel free to let us know your thoughts on these matters!

Another interesting approach is something like the autoloader of Shoelace where the DOM is observed and the component module is loaded at the time of element connection to the DOM.
https://shoelace.style/getting-started/installation#cdn-installation-easiest

**Comment by cjpillsbury** [comment]
should we have as part of the goal here support of `import 'media-chrome/menu'`, a la the new react exports/entry points?
